docs/design/adrs/2024-04-08-use-pull-throughout-for-ingestion.md

```diff
@@ -0,0 +1,47 @@
+# Use Pull approach(Integrated Vectorization) throughout for ingestion
+
+* **Status** - proposed
+* **Proposer:** @komalgrover
+* **Date:** 2024-04-08
+* **Technical Story:** [https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/321](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/321)
+
+## Context and Problem Statement
+
+There are two ways to ingest data to the search index i.e using Push based approach or a Pull based approach. This repository is currently using Push based approach in which the developer has more control on how the index is created and what values are put into the index.
+We want to add Integrated vectorization (Pull based approach) and give the flexibility to the user to use either of the two approaches. The purpose of this ADR is to document the approach taken to be able to deploy Pull based approach.
+
+## Decision Drivers
+* Ease of deployment
+* Ability to convey the concept of Integrated Vectorization (Pull based approach)
+
+## Considered Options
+* Using Pull approach throughout
+* Using Pull for documents and push for URL embeddings
+
+## Decision outcome
+* Using Pull approach throughout, as it meets all our requirements.
+
+## Pros and cons of each of the above options
+
+### Using Pull approach throughout
+
+Ask the user at the time of deployment on which approach to take. If the user selects Integrated Vectorization(IV), the resources for IV are created during the infra deployment through [bicep](https://learn.microsoft.com/en-us/azure/search/search-get-started-bicep?tabs=CLI) or REST APIs wrapped in script. Indexer can be scheduled to run every 5 min and all the chunking, vectorization process happens internally and data gets indexed.
```

Comment:
Can we make the indexer interval configurable rather than hardcode to 5 minutes?

Reply:
No longer needed for the indexer to be configurable, as it will run on every document upload